eZ/Bundle/EzPublishIOBundle/Migration/FileListerRegistry/IdentifierBased.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+/**
+ * File containing the IdentifierBased class.
+ *
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace eZ\Bundle\EzPublishIOBundle\Migration\FileListerRegistry;
+
+use eZ\Bundle\EzPublishIOBundle\Migration\FileListerRegistry;
+use eZ\Publish\Core\Base\Exceptions\NotFoundException;
+
+/**
+ * A registry of FileListerInterfaces that uses an identifier string to identify the lister.
+ */
+final class IdentifierBased implements FileListerRegistry
```

Comment:
I think this gives me a headache :) ConfigurableRegistry works if I remove the constructor from the interface (which I should, and just did). I'd say Configurable is a more important aspect than the internal array, so I'll go for that.